src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -2159,5 +2161,163 @@ public unsafe void M2()
                 //             await Task.Delay(2);
                 Diagnostic(ErrorCode.ERR_AwaitInUnsafeContext, "await Task.Delay(2)").WithLocation(33, 13));
         }
+
+        [Fact, WorkItem(16167, "https://github.com/dotnet/roslyn/issues/16167")]
+        public void Bug16167()
+        {
+            var text = @"
+class C
+{
+    public static void Main(int arg)
+    {
+        void Local1(bool b = M(arg is int z1, z1), int s1 = z1) {}
```

Comment:
>M(arg is int z1, z1) [](start = 29, length = 20)

We should test that locals and local functions declared within containing block are in scope inside default values. Semantic model should be tested for this as well: Lookup, GetSymbolInfo, etc.